spec.html

```diff
@@ -27574,8 +27574,10 @@ <h1>Runtime Semantics: DateString ( _tv_ )</h1>
             1. Let _weekday_ be the Name of the entry in <emu-xref href="#sec-todatestring-day-names"></emu-xref> with the Number WeekDay(_tv_).
             1. Let _month_ be the Name of the entry in <emu-xref href="#sec-todatestring-month-names"></emu-xref> with the Number MonthFromTime(_tv_).
             1. Let _day_ be the String representation of DateFromTime(_tv_), formatted as a two-digit decimal number, padded to the left with a zero if necessary.
-            1. Let _year_ be the String representation of YearFromTime(_tv_), formatted as a decimal number of at least four digits, padded to the left with zeroes if necessary.
-            1. Return the string-concatenation of _weekday_, the code unit 0x0020 (SPACE), _month_, the code unit 0x0020 (SPACE), _day_, the code unit 0x0020 (SPACE), and _year_.
+            1. Let _yv_ be YearFromTime(_tv_).
+            1. If _yv_ &ge; 0, let _yearSign_ be the empty string; otherwise, let _yearSign_ be `"-"`.
+            1. Let _year_ be the String representation of abs(_yv_), formatted as a decimal number of at least four digits, padded to the left with zeroes if necessary.
```

Comment:
i wonder if it might be useful here to extract out and share the internal logic for https://tc39.github.io/ecma262/#sec-string.prototype.padstart, such that you could call `PadStart(formatted as a decimal number of at least four digits, 4, '0')` or something?

(it'd be fine not to do that in this PR, ofc)

Reply:
I agree, and contemplated doing so in this PR but ultimately decided not to. I think a followup should replace all use of "formatted as a decimal number of at least $N digits, padded to the left with zeroes if necessary" with logic extracted from String.prototype.padStart.